### PR TITLE
Docs for function in ocp_nlp interface and ocp_qp interface 

### PR DESCRIPTION
**Pull request by jkoendev**
First set of function  documentation. There are still some TBD and TBC

### CODE REVIEW

File: acados/ocp_nlp/ocp_nlp_common.h [hunk 1]
```diff
@@ -126,30 +127,68 @@ int ocp_nlp_dims_calculate_size(void *config);
 ocp_nlp_dims *ocp_nlp_dims_assign_self(int N, void *raw_memory);
 //
 ocp_nlp_dims *ocp_nlp_dims_assign(void *config, void *raw_memory);
-//
+
+/// Sets the dimension of optimization variables
+/// (states, constrols, algebraic variables, slack variables).
+///
+/// \param config_ The configuration object.
```
Comment (zanellia) [inline acados/ocp_nlp/ocp_nlp_common.h]: maybe let's call it struct?

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 2]
```diff
@@ -84,20 +90,37 @@ typedef enum
...
     ocp_nlp_constraints_t *nlp_constraints;
+
+    /// Horizon length.
     int N;
-} ocp_nlp_plan;
 
+} ocp_nlp_plan;
 
 
+/// Structure to store state of the non-linear programming solver
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h]: ... to store the "configuration" of ...

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 3]
```diff
@@ -108,81 +131,203 @@ typedef struct
...
+/// Sets the function pointers to the dynamics functions for the given stage.
+///
+/// \param config The configuration object.
+/// \param dims The dimensions object.
+/// \param in The inputs object.
+/// \param stage Stage number.
+/// \param fun_type The name of the function type, either impl_ode_fun,
+///     impl_ode_fun_jac_x_xdot, impl_ode_jac_x_xdot_u (TBC)
+/// \param fun_ptr Function pointer to the dynamics function.
 int ocp_nlp_dynamics_model_set(ocp_nlp_config *config, ocp_nlp_dims *dims, ocp_nlp_in *in,
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h:201]: we should also add a \return to all function calls

File: acados/ocp_nlp/ocp_nlp_common.h [hunk 4]
```diff
@@ -126,30 +127,68 @@ int ocp_nlp_dims_calculate_size(void *config);
...
 typedef struct
 {
-    double *Ts;  // length of sampling intervals
+    /// Length of sampling intervals/timesteps.
+    double *Ts;
 
+    /// Pointers to cost functions (TBC).
     void **cost;
+
+    /// Pointers to dynamics functions (TBC).
```
Comment (zanellia) [inline acados/ocp_nlp/ocp_nlp_common.h:186]: pointer to dynamics "struct"? [same below]

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 5]
```diff
@@ -64,16 +64,22 @@ typedef enum
 } ocp_nlp_dynamics_t;
 
 
-
+/// Bound types
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h]: "Constraint types" would be better

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 6]
```diff
@@ -84,20 +90,37 @@ typedef enum
 } ocp_nlp_reg_t;
 
 
+/// Structure to store the configuration of a non-linear program
 typedef struct
 {
+    /// Qp solver configuration.
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h]: "QP" would be better.

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 7]
```diff
@@ -108,81 +131,203 @@ typedef struct
...
-// TODO remove and use ocp_nlp_dynamics_model_set instead !!!
+
+
+/// Deprecated
+/// TODO remove and use ocp_nlp_dynamics_model_set instead !!!
 int nlp_set_discrete_model_in_stage(ocp_nlp_config *config, ocp_nlp_in *in, int stage,
                                     void *fun_ptr);
-//
+
+/// Sets the function pointers to the cost functions for the given stage.
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h:210]: not only function pointers

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 8]
```diff
@@ -108,81 +131,203 @@ typedef struct
...
+///
+/// \param config The configuration object.
+/// \param dims The dimensions object.
+/// \param in The inputs object.
+/// \param stage Stage number.
+/// \param fun_type The name of the function type, either nls_res_jac,
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h]: "field"

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 9]
```diff
@@ -108,81 +131,203 @@ typedef struct
...
+///     y_ref, W (TBC)
+/// \param fun_ptr Function pointer to the cost function.
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h]: "value"

File: interfaces/acados_c/ocp_nlp_interface.h [hunk 10]
```diff
@@ -108,81 +131,203 @@ typedef struct
...
 int ocp_nlp_cost_model_set(ocp_nlp_config *config, ocp_nlp_dims *dims, ocp_nlp_in *in,
                            int stage, const char *field, void *value);
-//
+
+
+/// Sets the function pointers to the constraints functions for the given stage.
```
Comment (zanellia) [inline interfaces/acados_c/ocp_nlp_interface.h:223]: same as above

File: interfaces/acados_c/ocp_qp_interface.h [hunk 11]
```diff
@@ -27,7 +27,22 @@ extern "C" {
 #include "acados/ocp_qp/ocp_qp_common.h"
 
 
-
+/// Qp solver types (Enumeration).
```
Comment (zanellia) [inline interfaces/acados_c/ocp_qp_interface.h]: "QP"

**Review by zanellia — COMMENTED** [review]
Hey @jkoendev thanks a lot for the good work on documentation ;) There are a few things to be fixed, but it's a very good starting point!

File: interfaces/acados_c/ocp_nlp_interface.h (hunk 3 above)
Comment (jkoendev) [reply interfaces/acados_c/ocp_nlp_interface.h:201]: i would leave out the \return if the function returns nothing, as it's just additional work to write them everywhere..

File: acados/ocp_nlp/ocp_nlp_common.h (hunk 4 above)
Comment (jkoendev) [reply acados/ocp_nlp/ocp_nlp_common.h:186]: this is where i am not really sure what it actually is but my guess was that it actually includes the function pointer to the dynamics function.

File: interfaces/acados_c/ocp_nlp_interface.h (hunk 7 above)
Comment (jkoendev) [reply interfaces/acados_c/ocp_nlp_interface.h:210]: what else? I mean the argument fun_ptr contains the function pointer that is set somewhere which is think is the main functionality of this function

**Comment by jkoendev** [comment]
thanks for the feedback i addressed the issues (which are not subject to discussion) in 6a56c8b497b1476520e7efe79fe482195b311671

File: interfaces/acados_c/ocp_nlp_interface.h (hunk 3 above)
Comment (zanellia) [reply interfaces/acados_c/ocp_nlp_interface.h:201]: this returns "status" though :)

Comment (jkoendev) [reply interfaces/acados_c/ocp_nlp_interface.h:201]: true :) i will remember and add this to the next pull request/iteration of documentation.

**Pull request merged**